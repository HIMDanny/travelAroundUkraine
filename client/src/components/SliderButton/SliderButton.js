import React from 'react';
import { styled, IconButton } from '@mui/material';
import PropTypes from 'prop-types';
import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';

const Button = styled((props) => <IconButton color="primary" disableRipple {... props}/>)(({ theme }) => ({
  position: 'absolute',
  height: '60px',
  width: '60px',
  backgroundColor: theme.palette.common.white,
  zIndex: '2',
}));

const SliderButton = ({ id, position }) => (
  <Button id={id} sx={position}>
    <ArrowForwardIosIcon />
  </Button>
);

SliderButton.propTypes = {
  id: PropTypes.string,
  position: PropTypes.object,
};

SliderButton.defaultProps = {
  id: '',
  position: {},
};

export default SliderButton;
