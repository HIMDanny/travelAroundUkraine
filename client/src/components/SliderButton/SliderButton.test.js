import { render } from '@testing-library/react';
import SliderButton from './SliderButton';

jest.mock('@mui/material/IconButton', () => ({ children }) => <button>{children}</button>);

describe('SliderButton snapshot test', () => {
  test('should SliderButton match snapshot', () => {
    const { asFragment } = render(<SliderButton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
