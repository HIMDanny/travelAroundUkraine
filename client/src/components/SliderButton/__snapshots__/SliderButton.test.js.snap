// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderButton snapshot test should SliderButton match snapshot 1`] = `
<DocumentFragment>
  <button>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowForwardIosIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
      />
    </svg>
  </button>
</DocumentFragment>
`;
